src/serac/physics/utilities/functional/tuple.hpp

```diff
@@ -0,0 +1,408 @@
+#pragma once
+
+#include <utility>
```

Comment:
We need some unit tests for these..

Reply:
there are unit tests for `serac::tuple` here: 
https://github.com/LLNL/serac/blob/feature/jonwong12/gpu_functional_comparisons/src/serac/physics/utilities/functional/tests/tuple_arithmetic_unit_tests.cpp
but they may not be exhaustive

Reply:
missed that. should be good enough. just wanted a basic level of correctness checking before it blows up in larger things.